pkg/instance/repository.go

```diff
@@ -110,19 +113,73 @@ func (r repository) Delete(id uint) error {
 	return r.db.Unscoped().Delete(&model.Instance{}, id).Error
 }
 
-func (r repository) FindByGroupNames(names []string, presets bool) ([]*model.Instance, error) {
-	var instances []*model.Instance
+const AdministratorGroupName = "administrators"
 
-	err := r.db.
+type GroupWithInstances struct {
+	Name      string
+	Hostname  string
+	Instances []*model.Instance
+}
+
+func (r repository) FindByGroups(groups []*models.Group, presets bool) ([]GroupWithInstances, error) {
+	groupsByName := make(map[string]*models.Group)
+	for _, group := range groups {
+		groupsByName[group.Name] = group
+	}
+	groupNames := maps.Keys(groupsByName)
```

Comment:
You could gather the `groupNames` in your loop 😄 Since you are already looping over all groups and this is the only usage of this dependency. `maps.Keys` will loop again.

I'll leave that up to you 👌🏻

Reply:
I think that would make it less readable and since it won't affect performance I'll leave it as is. Thanks for the suggestion though :-)